packages/flutter/lib/src/material/button.dart

```diff
@@ -354,9 +365,22 @@ class _RawMaterialButtonState extends State<RawMaterialButton> {
   Widget build(BuildContext context) {
     final Color effectiveTextColor = MaterialStateProperty.resolveAs<Color>(widget.textStyle?.color, _states);
     final ShapeBorder effectiveShape =  MaterialStateProperty.resolveAs<ShapeBorder>(widget.shape, _states);
+    final Offset densityAdjustment = widget.density.adjustment;
+    final BoxConstraints densityConstraints = widget.constraints.copyWith(
```

Comment:
Would the variable name `adjustedConstraints` be more appropriate here? In the event that `widget.constraints.minWidth` is null, this variable would not have anything to do with density.